File: 0019-remove-nth-node-from-end-of-list/0019-remove-nth-node-from-end-of-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode removeNthFromEnd(ListNode head, int n) {
        ListNode temp2 = head;
        int l=0;
        while(temp2!=null){
            temp2=temp2.next;
            l+=1;
        }
        if(l==n){
            ListNode newHead = head.next;
            return newHead;
        }
        int res = l-n;
        ListNode temp=head;
        while(temp!=null){
            res--;
            if(res==0){
                break;
            }
            temp=temp.next;
        }
        ListNode del = temp.next;
        temp.next = temp.next.next;
        return head;
    }
}

File: 0019-remove-nth-node-from-end-of-list/0019-remove-nth-node-from-end-of-list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def removeNthFromEnd(self, head: Optional[ListNode], n: int) -> Optional[ListNode]:
        '''if head.next==None and n==1:
            return None'''
        l = 0
        cur = head
        while cur!=None:
            l+=1
            cur = cur.next
        if l==n:
            return head.next
        p = 0
        prev = ListNode(0)
        prev.next = head
        cur = head
        while p<(l-n):
            p+=1
            prev = cur
            cur = cur.next
        prev.next = cur.next
        cur.next = None
        return head
        
        